crates/bevy_ui/src/keyboard_navigation.rs

```diff
@@ -0,0 +1,173 @@
+use crate::{FocusPolicy, Interaction, UiStack};
+use bevy_a11y::Focus;
+use bevy_ecs::prelude::Component;
+use bevy_ecs::system::{Local, Query, Res, ResMut};
+use bevy_ecs::{change_detection::DetectChangesMut, entity::Entity, query::WorldQuery};
+use bevy_input::{prelude::KeyCode, Input};
+use bevy_reflect::{FromReflect, Reflect};
+use bevy_render::view::ComputedVisibility;
+
+/// A component that represents if a UI element is focused.
+#[derive(Reflect, FromReflect, Component, Copy, Clone, Debug, Eq, PartialEq)]
+pub enum FocusedState {
+    /// Nothing has happened
+    None,
+    /// Entity is focused
+    Focus {
+        /// Focus has been reached through keyboard navigation and so a focus style should be displayed.
+        /// This is similar to the `:focus-visible` pseudo-class in css.
+        focus_visible: bool,
+    },
+}
```

Comment:
The focus visible state is available as part of the `Focus` resource. I added it to the component so that it is more discoverable for users who just want to style the focus visible state. I personally think that matching against `FocusedState::Focus { focus_visible: bool }` further increases this discoverability, however I'm open to your opinion.